lib/control-connection.js

```diff
@@ -439,8 +449,21 @@ class ControlConnection extends events.EventEmitter {
         hostIterator = await promiseUtils.newQueryPlan(this._profileManager.getDefaultLoadBalancing(), null, null);
       }
 
-      this.connection = this._borrowAConnection(hostIterator);
+      try {
```

Comment:
This mirrors logic in init(), but in that function we only resolve contact point addresses if SNI is not in use.  Perhaps we should impose a similar constraint here?  It probably doesn't _hurt_ anything to do the resolution in all cases but it's not very useful in the SNI case.